### PR TITLE
Search-In-Worspace: no more different substrings/duplicates in search result

### PR DESCRIPTION
**Pull request by jbicker**
#### What it does
It solves the bug described here: #7282.
TL;DR: Sometimes one had different substrings in the search results and so one line shown more than once.

Fixes eclipse-theia/theia#7282

#### How to test
Type into the search box a search term and keep typing while the search is performed in the background. E.g. search for “ContainerModule”. Stop typing and wait for the result list. 
There shouldn't be any duplicates anymore.

#### Review checklist

- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by kittaakos** [comment]
> It solves the bug described [here](#7282).

The link does not work.

**Comment by jbicker** [comment]
> > It solves the bug described [here](#7282).
> 
> The link does not work.

Thanks! Obviously I expected too much and tested the comment too less.

File: packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx [hunk 1]
```diff
@@ -212,10 +212,15 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
...
         const cancelIndicator = this.cancelIndicator;
         const token = this.cancelIndicator.token;
+        const progress = await this.progressService.showProgress({ text: `search: ${searchTerm}`, options: { location: 'search' } });
         token.onCancellationRequested(() => {
             this.changeEmitter.fire(this.resultTree);
+            progress.cancel();
+            if (searchId) {
+                this.searchService.cancel(searchId);
+            }
+            this.cancelIndicator = undefined;
```
Comment (kittaakos) [inline packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx:222]: [minor]: I think `this.changeEmitter.fire(this.resultTree);` should come after this line. When everything has been canceled and unset, you can fire the event. What do you think?

Comment (jbicker) [reply packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx:221]: done
